src/lib/protocols/credentials/CredentialService.ts

```diff
@@ -0,0 +1,66 @@
+import { v4 as uuid } from 'uuid';
+import { EventEmitter } from 'events';
+import { CredentialRecord } from '../../storage/CredentialRecord';
+import { Repository } from '../../storage/Repository';
+import { Wallet } from '../../wallet/Wallet';
+import { CredentialOfferMessage, CredentialPreview, Attachment } from './messages/CredentialOfferMessage';
+import { InboundMessageContext } from '../../agent/models/InboundMessageContext';
+import { CredentialState } from './CredentialState';
+
+export enum EventType {
+  StateChanged = 'stateChanged',
+}
+
+export class CredentialService extends EventEmitter {
+  private wallet: Wallet;
+  private credentialRepository: Repository<CredentialRecord>;
+
+  public constructor(wallet: Wallet, credentialRepository: Repository<CredentialRecord>) {
+    super();
+    this.wallet = wallet;
+    this.credentialRepository = credentialRepository;
+  }
+
+  public async createCredentialOffer({
+    credDefId,
```

Comment:
```suggestion
    credentialDefinitionId,
```

My understanding was that we will use `credDefId` only for typing indy-sdk and use the full version everywhere else in the framework

Reply:
Ok 👍  I probably missed that or was lazy to replace it :)